src/NodaTime.Demo/AnnualDateDemo.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright 2019 The Noda Time Authors. All rights reserved.
+// Use of this source code is governed by the Apache License 2.0,
+// as found in the LICENSE.txt file.
+
+using NUnit.Framework;
+using System;
+namespace NodaTime.Demo
+{
+    class AnnualDateDemo
+    {
+        [Test]
+        public void CompareTo()
+        {
+            var earlierDate = new AnnualDate(2, 23);
```

Comment:
I think it might be useful to name these `february23rd` and `august17th` for simplicity - ditto later.

That way, we don't end up with variables called `date` which don't really represent a full date, too.